zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/listeners/task/TaskListenerTest.java

```diff
@@ -162,6 +162,62 @@ public void shouldAssignUserTaskAfterAllAssignmentTaskListenersAreExecuted() {
             Assertions.assertThat(userTask).hasAssignee("me").hasAction("my_assign_action"));
   }
 
+  @Test
+  public void shouldRejectUserTaskAssignmentWhenTaskListenerRejectsTheOperation() {
+    // given
+    final long processInstanceKey =
+        createProcessInstance(
+            createUserTaskWithTaskListeners(ZeebeTaskListenerEventType.assignment, LISTENER_TYPE));
+
+    ENGINE.userTask().ofInstance(processInstanceKey).assign();
+    ENGINE
+        .job()
+        .ofInstance(processInstanceKey)
+        .withType(LISTENER_TYPE)
+        .withResult(new JobResult().setDenied(true))
+        .complete();
+
+    // then: ensure that `REJECT_TASK_LISTENER` and `ASSIGNMENT_DENIED`
+    // are written after `ASSIGNING` event
+    assertUserTaskIntentsSequence(
+        UserTaskIntent.ASSIGNING,
+        UserTaskIntent.DENY_TASK_LISTENER,
+        UserTaskIntent.ASSIGNMENT_DENIED);
```

Comment:
❌ This test should show that the `assignee` did not change in addition to the intent sequence. Otherwise, our tests don't shows the full behavior we want to provide.

Reply:
Good catch, I've modified the test to verify the assignee as well, will be committing the change shortly 👍